boefjes/boefjes/plugins/kat_nikto/boefje.json

```diff
@@ -0,0 +1,14 @@
+{
+  "id": "nikto",
+  "name": "Nikto",
+  "description": "Uses Nikto",
+  "consumes": [
+    "IPAddressV4",
+    "IPAddressV6",
+    "Hostname"
```

Comment:
I think it makes more sense to start Nikto on Ipports that we know are talking http/https.
This way we avoid nikto on services that are not webservers, or even better:

It looks like there's also a way to differentiate between different hostnames on the same IP, so scanning on websites could also make sense as that would provide the best data-set including any differences between ipv4 and ipv6 addresses or the same host. Use the cli argument -host+ for this.

Reply:
I have looked more into it and think using the OOI `HostnameHTTPURL` would be perfect. It has a hostname, port, and scheme which has everything. I was not able to find information from `IpPort` whether it was a webserver or not (unless you meant looking at the port's number). 

`-h+` is supposed to be an alias for `-host+` although this is not specified in [nikto's wiki](https://github.com/sullo/nikto/wiki/Annotated-Option-List). I will make sure to use `-nolookup`.

Reply:
The boefje now makes use of `HostnameHTTPURL` which tells the boefje whether its HTTP or HTTPS and contains a hostname as well. 

In theory `IPAddressHTTPURL` could also be used, I plan to add this once the boefje is satisfactory.